lib/utils/makeValidator.js

```diff
@@ -3,21 +3,116 @@
 const jsonschema = require('jsonschema');
 const links = require('./links');
 const functionalConstraints = require('../functional-constraints');
+const { flattenDeep } = require('lodash');
+
+const ambiguousTypes = ['anyOf', 'oneOf', 'allOf'];
 
 const makeLinks = (error, makerFunc) => {
-  if (typeof error.schema == 'string') {
+  // debugger; // eslint-disable-line no-debugger
+  if (typeof error.schema === 'string') {
     return [makerFunc(error.schema)];
   }
   if (
-    ['anyOf', 'oneOf', 'allOf'].indexOf(error.name) !== -1 &&
+    ambiguousTypes.includes(error.name) &&
     error.argument &&
     error.argument.length
   ) {
-    return error.argument.map(makerFunc);
+    // no way to know what the subschema was, so don't create links for it
+    return error.argument
+      .map(s => (s.includes('subschema') ? '' : makerFunc(s)))
+      .filter(Boolean);
   }
   return [];
 };
 
+const removeFirstAndLastChar = s => s.slice(1, -1);
+// always return a string
+const makePath = (path, newSegment) =>
+  (path ? [path, newSegment].join('.') : newSegment) || '';
+
+const processBaseError = (err, path) => {
+  const completePath = makePath(path, err.property).replace(
+    /\.instance\./g,
+    '.'
+  );
```

Comment:
I see there are some cases where the path ends with `.instance`. Maybe we can change it to something like:
```js
  const completePath = _.trimEnd(
    makePath(path, err.property).replace(/\.instance($|\.)/g, '.'),
    '.'
  );
```

Reply:
yeah, good call. If you've still got it, it would be great to get an input + schema that generated that so I can write a test against it

Reply:
One example is when you validate a primitive value against `FieldSchema`. Like `inputFields: [123]`.